### PR TITLE
[docs] [ENG-8506] Add new on_cancel hook to the docs

### PR DESCRIPTION
**Pull request by radoslawkrzemien**
Updated the build lifecycle hooks docs with the new on_cancel hook, with the updated package.json example

# Why

After adding the cleanup phase to the worker whenever the build is terminated before it can finish, a new lifecycle hook (on_cancel) is being added as well. Updating the docs to mention it

See: https://linear.app/expo/issue/ENG-8506/attempt-to-upload-xcode-logs-when-build-fails-for-unexpected-reason-or

# How

On the page regarding build lifecycle hooks a new point is added for the on_cancel hook, and the example given below is also updated

# Test Plan

Run `yarn run dev` and checked the results
Screenshot 2023-06-28 at 11 53 43

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/build-reference/npm-hooks.mdx [hunk 1]
```diff
@@ -12,6 +12,7 @@ There are five EAS Build lifecycle npm hooks that you can set in your **package.
 - `eas-build-on-success` - this hook is triggered at the end of the build process if the build was successful.
 - `eas-build-on-error` - this hook is triggered at the end of the build process if the build failed.
 - `eas-build-on-complete` - this hook is triggered at the end of the build process. You can check the build's status with the `EAS_BUILD_STATUS` environment variable. It's either `finished` or `errored`.
+- `eas-build-on-cancel` - this hook is triggered if the build gets canceled before it can finish. It is executed at the end of cleanup phase, just before the build worker gets terminated.
```
Comment (szdziedzic) [inline docs/pages/build-reference/npm-hooks.mdx:15]:
```suggestion
- `eas-build-on-cancel` - this hook is triggered if the build is canceled.
```

I think this explanation should be enough. It is less confusing and more precise in my opinion.

What do you think?

Comment (radoslawkrzemien) [reply docs/pages/build-reference/npm-hooks.mdx:15]: Sure, no strong opinion either way on my side 👍

**Review by brentvatne — APPROVED** [review]

**Review by szdziedzic — APPROVED** [review]
Looks good, this should be merged after hook works fine and is released in Turtle v2 repo

**Comment by radoslawkrzemien** [comment]
Thanks for the approves, waiting on the hooks changes to be deployed first before deploying this

**Pull request merged**